conf/evolutions/reversions/125-allow-dollar-in-layer-names.sql

```diff
@@ -0,0 +1,11 @@
+START TRANSACTION;
+
+-- This reversion might take a while because it needs to search in all annotation layer names for '$' and replace it with ''
+do $$ begin ASSERT (select schemaVersion from webknossos.releaseInformation) = 125, 'Previous schema version mismatch'; end; $$ LANGUAGE plpgsql;
+
+UPDATE webknossos.annotation_layers SET name = regexp_replace(name, '\$', '', 'g') WHERE name ~* '\$';
```

Comment:
@fm3 There is a problem here with the reversion. Removing `$` from existing annotation layer names makes them very likely invalid / not findable in the fossil db.
Therefore, I'd say that this PR is not really revertable? Otherwise one needs to fix the data in the fossilDB. Or at least the schema cannot be rolled back, because rolling back the `annotation_layers_name_check` would require no annotation layer names with `$` in their name :thinking: 

At least that's a effect I observed while checking the reversion. An annotation with an annotation layer without fallback that included `$` was suddenly empty (if I remember correctly that I did some annotations in that layer).

Reply:
the layer names are not used for lookup in the fossildb, only the tracingid is used for that. I think renaming annotation layers should be fairly harmless.

On the other hand, if the fallback segmentation name changes, that may lead to issues.

Not sure what you saw there :thinking: Could you try again if you can reproduce this? After a page refresh the data should be ok.

Reply:
Ok I gave this another investigation. I could not reproduce the error I remembered. Maybe I messed up somewhere.
The only error I could produce was with a exactly that:
> On the other hand, if the fallback segmentation name changes, that may lead to issues.

See: 
![image](https://github.com/user-attachments/assets/f804f895-84f4-403b-bb58-9bb8282d0956)
Before executing the reversion the name of the segmentation fallback layer & the volume annotation layer was `segmentation$dollar`. After the reversion the annotation layer name was `segmentationdollar` but the fallback layers name stayed at `segmentation$dollar`. Then I manually renamed the fallback segmentation layer to `segmentationdollar` in the datasource and folder structure. => viewing the ds works but the volume segmentation layer is now detached from the fallback layer (as shown in the screenshot). 
But I'd say that this is an expected problem due to the renaming of the underlying segmentation layer? 

In short: I think I was just wrong here, claiming that the reversion might cause problems. Sorry :see_no_evil:

Reply:
That’s good news then :) Yes, that sounds plausible